File: src/main/java/hello/hellospring/repository/MemberRepository.java
package hello.hellospring.repository;

import hello.hellospring.domain.Member;

import java.util.List;
import java.util.Optional;

public interface MemberRepository {
    Member save(Member member);
    Optional<Member> findById(Long id);         // Optional <- Java8에 들어간 기능.
    Optional<Member> findById(String name);     //          findByName이나 findById를 했을 때 null일 수 있음
    List<Member> findAll();                     //          null을 처리하는 방법이라 생각하면 됨
}

File: src/main/kotlin/hello/hellospring/repository/MemberRepository.kt
package hello.hellospring.repository

import hello.hellospring.domain.Member
import hello.hellospring.dto.MemberDto
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface MemberRepository {

    fun save(member: MemberDto): Member

    fun findById(id: Long?): Optional<Member>

    fun findByName(name: String?): Optional<Member>

    fun findAll(): List<Member>
}